### PR TITLE
test: add vm-level fuzzing

### PR DESCRIPTION
**Pull request by matklad**
Finally got to publishing fuzzing infra we have been using in ad-hoc ways before. This just runs the contracts to make sure that VM doesn't crash. The code is setup to easily compare execution with different VM Kinds. 

Note that at the moment fuzzing *does not* generally exercise host function calls. I need to finish https://github.com/bytecodealliance/wasm-tools/pull/286 for that :sweat_smile:

### CODE REVIEW

File: runtime/near-vm-runner/fuzz/fuzz_targets/runner.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                }
+            }
+            _ => (),
+        }
+    }
+    None
+}
+
+fn create_context(input: Vec<u8>) -> VMContext {
+    VMContext {
```
Comment (nagisa) [inline runtime/near-vm-runner/fuzz/fuzz_targets/runner.rs:70]: Aren't some of these potentially controlled by the user? Would be ideal if the user-influenced parts were `Arbitrary` too. Potentially as a separate fuzz target?

**Review by nagisa — APPROVED** [review]
This seems reasonable to me. I'm amazed how well `arbitrary` has caught on that there's even an implementation of `Arbitrary` to generate wasm modules!

File: runtime/near-vm-runner/fuzz/fuzz_targets/runner.rs (hunk 1 above)
Comment (matklad) [reply runtime/near-vm-runner/fuzz/fuzz_targets/runner.rs:70]: Yeah, that's a good idea. Let me issue this... #5465

**Comment by mina86** [comment]
pytest and nightly changes LGTM

File: pytest/lib/cargo_fuzz.py [hunk 2]
```diff
@@ -1,9 +1,26 @@
...
 import os
-import subprocess
 import sys
+import subprocess
+
+
+def run(dir: str, fuzz_target: str) -> int:
+    args = ('cargo', 'fuzz', 'run', fuzz_target, '--', '-len_control=0'
+            '-prefer_small=0', '-max_len=4000000', '-rss_limit_mb=10240')
+    os.environ['RUSTC_BOOTSTRAP'] = '1'
```
Comment (olonho) [inline pytest/lib/cargo_fuzz.py:10]: Hmm, why it's needed?

File: runtime/near-vm-runner/fuzz/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+[package]
+name = "near-vm-runner-fuzz"
+version = "0.0.0"
+authors = ["Automatically generated"]
```
Comment (olonho) [inline runtime/near-vm-runner/fuzz/Cargo.toml:4]: Is it?

**Review by olonho — APPROVED** [review]

File: runtime/near-vm-runner/fuzz/Cargo.toml (hunk 3 above)
Comment (matklad) [reply runtime/near-vm-runner/fuzz/Cargo.toml:4]: It is. Though as `authors` field is now optional, it's better to remove it here (and upstream in the generator itself as well, cc @nagisa ).

File: pytest/lib/cargo_fuzz.py (hunk 2 above)
Comment (matklad) [reply pytest/lib/cargo_fuzz.py:10]: To spell this implicitly, but butting fuzzing into our CI we make it our responsibility to deal with any breakage related to compiler version upgrades. This in contrast to all over Rust functionality we use -- if something stable&documented breaks with an upgrade, we can and should complain upstream about that.

**Pull request merged**